runner.go

```diff
@@ -415,8 +422,13 @@ func (runner *Runner) workerDone(info doneInfo) {
 		return
 	}
 	if info.err != nil {
+		errStr := info.err.Error()
+		if errWithStack, ok := info.err.(panicError); ok && errWithStack.Panicked() {
```

Comment:
Isn't sufficient just to assert that err is of type errWithStackTrace?

Reply:
That type is in a different package. Plus, StackTrace() []string is also use by the juju/errors package, which would mean all juju errors would have a stack trace.